### ISSUE TITLE
missing functions to use editline as drop-in replacement for readline with gdb 7.6

### ISSUE THREAD

**Issue by rofl0r** [troglobit/editline#9]
```
 undefined reference to `history_base'
 undefined reference to `history_expand'
 undefined reference to `history_get'
 undefined reference to `history_is_stifled'
 undefined reference to `history_length'
 undefined reference to `max_input_history'
 undefined reference to `rl_add_defun'
 undefined reference to `rl_already_prompted'
 undefined reference to `rl_callback_handler_install'
 undefined reference to `rl_callback_handler_remove'
 undefined reference to `rl_callback_read_char'
 undefined reference to `rl_completer_quote_characters'
 undefined reference to `rl_completer_word_break_characters'
 undefined reference to `rl_completion_entry_function'
 undefined reference to `rl_completion_word_break_hook'
 undefined reference to `rl_filename_completion_function'
 undefined reference to `rl_get_previous_history'
 undefined reference to `rl_get_screen_size'
 undefined reference to `rl_newline'
 undefined reference to `rl_pre_input_hook'
 undefined reference to `rl_redisplay'
 undefined reference to `rl_set_screen_size'
 undefined reference to `rl_terminal_name'
 undefined reference to `stifle_history'
 undefined reference to `tilde_expand'
 undefined reference to `unstifle_history'
 undefined reference to `where_history'
```
gdb 7.6 seems to be the most elaborate readline user in sabotage linux, but it can be tricked[0] to link against libedit instead of readline, and i hoped as well against editline...

[0] https://github.com/sabotage-linux/sabotage/blob/master/pkg/libedit#L21

**Comment by troglobit** [comment]
Wow, glad to see someone trying to use editline as a drop-in replacement for GNU readline! :-)

The editline library was never intended as a GNU readline() replacement. And although I've considered myself implementing the necessary bits and bobs for GDB to work, I never found the time during my ten years maintaining this project.

However, I would greatly appreciate any and all patches to help reach that target for editline, since we all would greatly benefit from a truly FREE readline() replacement!

**Comment by rofl0r** [comment]
yeah, having a slim drop-in replacement for the utterly bloated readline would be very nice, since it's unlikely that the GNU guys will ever switch to something else, and libedit is also rather bloated.

**Comment by troglobit** [comment]
Yep, that'd be nice. Plain termios support for all, I say :-)

**Comment by troglobit** [comment]
Update: edited original report slightly so it can be used as a TODO list.  Some functions have now been implemented, checked.

**Comment by troglobit** [comment]
Update: two new functions/callbacks added.

**Comment by firasuke** [comment]
Any updates on this?

**Comment by troglobit** [comment]
Not much no. 

Just to be clear, the point of editline is not to be a replacement for alll functions in readline. If we can replace just enough to be a suitable alternative, fine, but if the size skyrockets and requirements on external libraries (not libc, but ncurses) sneak in, well then we're off the mark completely.  So for me, this issue serves only as a reminder; to check against when refactoring/adding features to see if the APIs listed here can be wrangled in there without much extra effort.

**Comment by firasuke** [comment]
I see.

Thanks for the update!
